src/shared/port-util.js

```diff
@@ -42,14 +42,12 @@ export function isMessageEqual(data, message) {
     return false;
   }
 
-  try {
-    return (
-      JSON.stringify(data, Object.keys(data).sort()) ===
-      JSON.stringify(message, Object.keys(message).sort())
-    );
-  } catch {
-    return false;
-  }
+  // We assume `JSON.stringify` cannot throw because `isMessage` verifies that
+  // all the fields we serialize here are serializable values.
+  return (
+    JSON.stringify(data, Object.keys(message).sort()) ===
+    JSON.stringify(message, Object.keys(message).sort())
+  );
```

Comment:
There is a risk here that future changes to the format of messages or the `isMessage` logic breaks this invariant. On the other hand this eliminates a code path which is currently never used.